discovery_engine_core/core/src/engine.rs

```diff
@@ -185,6 +185,12 @@ where
         bincode::serialize(&state_data).map_err(|err| Error::Serialization(err.into()))
     }
 
+    /// Updates the markets configuration.
+    pub async fn set_markets(&mut self, markets: Vec<Market>) {
+        let mut mkts = self.config.markets.write().await;
+        *mkts = markets;
```

Comment:
this could be done without the intermediate binding, but both ways are ok i guess.
```suggestion
        *self.config.markets.write().await = markets;
```